HyperVExtension/BuildDevSetupAgentHelper.ps1

```diff
@@ -0,0 +1,109 @@
+Param(
+    [string]$Platform = "x64",
+    [string]$Configuration = "debug",
+    [string]$VersionOfSDK,
+    [string]$SDKNugetSource,
+    [string]$Version,
+    [string]$BuildStep = "all",
+    [string]$AzureBuildingBranch = "main",
+    [switch]$IsAzurePipelineBuild = $false,
+    [switch]$BypassWarning = $false,
+    [switch]$Help = $false
+)
+
+$StartTime = Get-Date
+
+if ($Help) {
+  Write-Host @"
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT License.
+
+Syntax:
+      BuildDevSetupAgentHelper.cmd [options]
+
+Description:
+      Builds DevSetupAgent.
+
+Options:
+
+  -Platform <platform>
+      Only build the selected platform(s)
+      Example: -Platform x64
+      Example: -Platform "x86,x64,arm64"
+
+  -Configuration <configuration>
+      Only build the selected configuration(s)
+      Example: -Configuration Release
+      Example: -Configuration "Debug,Release"
+
+  -Help
+      Display this usage message.
+"@
+  Exit
+}
+
+if (-not $BypassWarning) {
+  Write-Host @"
+This script is not meant to be run directly. To build DevSetupAgent, please run the following from the root directory:
+build -BuildStep "DevSetupAgent"
+"@ -ForegroundColor RED
+  Exit
+}
+
+$isAdmin = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] 'Administrator')
+
+$ErrorActionPreference = "Stop"
+
+$msbuildPath = &"${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe
+if ($IsAzurePipelineBuild) {
+  $nugetPath = "nuget.exe";
+} else {
+  $nugetPath = (Join-Path $env:Build_RootDirectory "build\NugetWrapper.cmd")
+}
+
+if (-not([string]::IsNullOrWhiteSpace($SDKNugetSource))) {
+  & $nugetPath sources add -Source $SDKNugetSource
+}
+
+Try {
+  $buildRing = "Dev"
+
+  if ($AzureBuildingBranch -ieq "release") {
+    $buildRing = "Stable"
+  } elseif ($AzureBuildingBranch -ieq "staging") {
+    $buildRing = "Canary"
+  }
+
+  $msbuildArgs = @(
+      ("HyperVExtension\DevSetupAgent.sln"),
+      ("/p:Platform="+$platform),
+      ("/p:Configuration="+$configuration),
+      ("/restore"),
+      ("/binaryLogger:DevSetupAgent.$platform.$configuration.binlog"),
+      ("/p:BuildRing=$buildRing")
+  )
+  if (-not([string]::IsNullOrWhiteSpace($VersionOfSDK))) {
+    $msbuildArgs += ("/p:DevHomeSDKVersion="+$env:sdk_version)
+  }
+
+  & $msbuildPath $msbuildArgs
+
+  $binariesOutputPath = (Join-Path $env:Build_RootDirectory "HyperVExtension\src\DevSetupAgent\bin\$Platform\$Configuration\net8.0-windows10.0.22000.0\win10-$Platform\*")
```

Comment:
When we I move the hyper-v extension to a top level "Extensions" folder we'll need to change these. I'll make a note of this

Reply:
Yes, unfortunately there are hardcoded paths in a couple of places. Perhaps we can improve this later with some kind of variable that can be used here and from VS. (And then we'd need not to forget to update it.). It would be nice to be able to ask Msbuild: "Give me output location for this project", but I haven't found a way to do it so far.